src/Microsoft.Framework.Configuration/ConfigurationKeyComparer.cs

```diff
@@ -8,14 +8,12 @@ namespace Microsoft.Framework.Configuration
 {
     public class ConfigurationKeyComparer : IComparer<string>
     {
-        private const char Separator = ':';
-
         public static ConfigurationKeyComparer Instance { get; } = new ConfigurationKeyComparer();
 
         public int Compare(string x, string y)
         {
-            var xParts = x?.Split(Separator) ?? new string[0];
-            var yParts = y?.Split(Separator) ?? new string[0];
+            var xParts = x?.Split(new string[]{ Constants.KeyDelimiter }, StringSplitOptions.RemoveEmptyEntries) ?? new string[0];
```

Comment:
What if we use
```csharp
var xParts = x?.Split(KeyDelimiter[0]).Where(i => i != string.Empty).ToArray() ?? new string[0];
```
semantically the key delimiter is character rather than array of characters